DokuwikiXMLExport.php

```diff
@@ -0,0 +1,194 @@
+use FINDOLOGIC\Export\Data\Description;
+use FINDOLOGIC\Export\Data\Price;
+use FINDOLOGIC\Export\Data\Url;
+use FINDOLOGIC\Export\Data\Ordernumber;
+use FINDOLOGIC\Export\Data\DateAdded;
+use FINDOLOGIC\Export\Data\Attribute;
+
+class DokuwikiXMLExport
+{
+    /**
+     * Generate the entire XML Export based on the Dokuwiki metadata.
+     *
+     * @param $start integer Determines the first item (offset) to be exported.
+     * @param $count integer Determines the interval size / number of items to be exported.
+     * @return string Returns the XML.
+     */
+    public function generateXMLExport($start, $count)
+    {
+        global $conf;
+
+        $Indexer = new Doku_Indexer();
+        $pagesAndDeletedPages = $Indexer->getPages();
+
+        $excludedPages = $this->splitConfigToArray($conf['plugin']['findologicxmlexport']['excludePages']);
+        $pages = null;
+        foreach ($pagesAndDeletedPages as $page){
+            if (p_get_metadata($page)['description']){ // Only get pages with content
+                if (!in_array($page, $excludedPages)){ // Exclude pages from config
+                    $pages[] = $page;
+                }
+            }
+        }
+        $exporter = Exporter::create(Exporter::TYPE_XML, $count);
+
+        $total = count($pages);
+
+        // If count is higher then total, set count=total to prevent an exception.
+        if ($count > $total) {
+            $count = $total;
+        }
+
+        // If count + start is higher then total, then something went totally wrong.
+        if (($count + $start) > $total) {
+            echo ('Call "\DokuwikiXMLExport::generateXMLExport" failed while trying to validate start and count values.');
+            return false;
+        }
+
+        $items = array();
+        for ($i = $start; $i < $start + $count; $i++) {
+            $item = $exporter->createItem($i);
+
+            $name = new Name();
+            $name->setValue($this->getName($pages, $i));
+            $item->setName($name);
+
+            $summary = new Summary();
+            $summary->setValue($this->getSummaryAndDescription($pages, $i));
+            $item->setSummary($summary);
+
+            $description = new Description();
+            $description->setValue($this->getSummaryAndDescription($pages, $i));
+            $item->setDescription($description);
+
+            $price = new Price();
+            $price->setValue('0.0');
+            $item->setPrice($price);
+
+            $Url = new Url();
+            $Url->setValue($this->getUrl($pages, $i));
+            $item->setUrl($Url);
+
+            $dateAdded = new DateAdded();
+            $dateAdded->setDateValue($this->getDateAdded($pages, $i));
+            $item->setDateAdded($dateAdded);
+
+            $item->addOrdernumber(new Ordernumber($this->getOrdernumber($pages, $i)));
+            $items[] = $item;
+
+            $attributeCategory = new Attribute('cat', $this->getAttributesCategory($pages, $i));
```

Comment:
`cat` is a magic string and should be replaced with a constant.

Reply:
TODO:

- Make constant where magic string `cat` gets set. Also add a doc for this to understand why this string is like this.